qcodes/dataset/sqlite_base.py

```diff
@@ -1363,29 +1507,113 @@ def add_meta_data(conn: sqlite3.Connection,
             raise e
 
 
-def get_user_version(conn: sqlite3.Connection) -> int:
+def get_user_version(conn: SomeConnection) -> int:
 
     curr = atomic_transaction(conn, 'PRAGMA user_version')
     res = one(curr, 0)
     return res
 
 
-def set_user_version(conn: sqlite3.Connection, version: int) -> None:
+def set_user_version(conn: SomeConnection, version: int) -> None:
 
     atomic_transaction(conn, 'PRAGMA user_version({})'.format(version))
 
 
 def get_experiment_name_from_experiment_id(
-        conn: sqlite3.Connection, exp_id: int) -> str:
+        conn: SomeConnection, exp_id: int) -> str:
     return select_one_where(
         conn, "experiments", "name", "exp_id", exp_id)
 
 
 def get_sample_name_from_experiment_id(
-        conn: sqlite3.Connection, exp_id: int) -> str:
+        conn: SomeConnection, exp_id: int) -> str:
     return select_one_where(
         conn, "experiments", "sample_name", "exp_id", exp_id)
 
 
-def get_run_timestamp_from_run_id(conn: sqlite3.Connection, run_id: int) -> int:
+def get_run_timestamp_from_run_id(conn: SomeConnection,
+                                  run_id: int) -> float:
     return select_one_where(conn, "runs", "run_timestamp", "run_id", run_id)
+
+
+def update_GUIDs(conn: SomeConnection) -> None:
+    """
+    Update all GUIDs in this database where either the location code or the
+    work_station code is zero to use the location and work_station code from
+    the qcodesrc.json file in home. Runs where it is not true that both codes
+    are zero are skipped.
+    """
+
+    log.info('Commencing update of all GUIDs in database')
+
+    cfg = qc.Config()
+
+    try:
+        location = cfg['GUID_components']['location']
+    except KeyError:
+        log.warning('No location information found in the config. '
+                    'Can not proceed.')
+        return
+    try:
+        work_station = cfg['GUID_components']['work_station']
+    except KeyError:
+        log.warning('No work_station information found in the config. '
```

Comment:
if config does not even have `GUID_components`, then the warning message will not be completely correct. Shall it be adjusted to smth like `GUID_components -> work_station`?

Reply:
for `location` (above) as well?

Reply:
It's not wrong; that information is not in the config. But it can actually never happen, since we ship QCoDeS with those settings.